src/components/transaction/TransactionTable.js

```diff
@@ -16,21 +17,36 @@ const TransactionTable = () => {
 
   if (transactions.length === 0) {
     return (
-      <div className='container center-align'>
+      <div className="container center-align">
         <h4>Enter a transaction!</h4>
       </div>
     );
   }
 
   console.log(transactions);
 
+  const validateInputs = (transactions) => {
+    for (let transaction of transactions) {
+      if (!validations.validateAlphaNumeric(transaction.service)) {
+        return transaction.displayDate;
+      }
+    }
+    return false;
+  };
+
   const onClick = () => {
-    setAssetTypes(transactions);
+    let date = validateInputs(transactions);
+    if (!date) {
+      setAssetTypes(transactions);
+    } else {
+      // set transactionState hasError to date or transaction that caused error
+      updateError(date);
```

Comment:
I named `updateError` since the function updates the portion of the transaction state that holds the error information, but I can see how `showError` may be more descriptive since that is the end result of calling this function once React detects the state change and updates the DOM.